service/src/test/java/bio/terra/workspace/service/workspace/AwsCloudContextConnectedTest.java

```diff
@@ -0,0 +1,179 @@
+      assertTrue(landingZoneOptional.isPresent());
+
+      LandingZone landingZone = landingZoneOptional.get();
+      assertNotNull(landingZone.getMetadata(), region + ": landingZone.metadata null");
+      assertNotNull(landingZone.getStorageBucket(), region + ": landingZone.storageBucket null");
+      assertNotNull(landingZone.getKmsKey(), region + ": landingZone.kmsKey null");
+      assertFalse(
+          landingZone.getNotebookLifecycleConfigurations().isEmpty(),
+          region + ": landingZone.notebookLifecycleConfigurations empty");
+
+      // Log details about the landing zone.
+      logMetadata(landingZone.getMetadata(), "Landing zone region: " + region);
+
+      // Perform a HeadBucket command to check storage bucket existence; this requires ListBucket
+      // permissions on the bucket, which the WSM role has.  This is a basic "Hello, world!" test
+      // that we've discovered the environment and assumed the TerraWorkspaceManager role.
+
+      String bucketName = landingZone.getStorageBucket().name();
+      HeadBucketRequest request = HeadBucketRequest.builder().bucket(bucketName).build();
+
+      // SdkClient classes should generally be used in try-with-resource blocks.
+      try (S3Client s3Client =
+          S3Client.builder().region(region).credentialsProvider(awsCredentialsProvider).build()) {
+        s3Client.headBucket(request);
+      } catch (Exception exception) {
+        fail(String.format("HeadBucket request failed: %s", exception), exception);
+      }
+
+      logger.info("Confirmed access to bucket {} in AWS Region {}", bucketName, region);
+    }
+  }
+
+  @Test
+  void createCloudContextTest() {
+    AwsCloudContext createdCloudContext =
+        awsCloudContextService.createCloudContext(
+            "flightId", WorkspaceFixtures.DEFAULT_SPEND_PROFILE_ID);
+    assertNotNull(createdCloudContext);
+
+    Metadata envMetadata = awsCloudContextService.discoverEnvironment().getMetadata();
+    AwsCloudContextFields contextFields = createdCloudContext.getContextFields();
+    assertNotNull(contextFields);
+    assertEquals(envMetadata.getMajorVersion(), contextFields.getMajorVersion());
+    assertEquals(envMetadata.getOrganizationId(), contextFields.getOrganizationId());
+    assertEquals(envMetadata.getAccountId(), contextFields.getAccountId());
+    assertEquals(envMetadata.getTenantAlias(), contextFields.getTenantAlias());
+    assertEquals(envMetadata.getEnvironmentAlias(), contextFields.getEnvironmentAlias());
+
+    CloudContextCommonFields commonFields = createdCloudContext.getCommonFields();
+    assertNotNull(commonFields);
+    assertEquals("flightId", commonFields.flightId());
+    assertThat(commonFields.spendProfileId(), equalTo(WorkspaceFixtures.DEFAULT_SPEND_PROFILE_ID));
+    assertEquals(WsmResourceState.CREATING, commonFields.state());
+    assertNull(commonFields.error());
+  }
+
+  @Test
+  void getLandingZoneTest() throws IOException {
```

Comment:
there's no startup cost (no beforeEach), can you split them up for readibility? 
getLandingZone_success
getLandingZone_noLandingZonRegion_returnsEmpty
getLandingZone_badCloudContext_throwsInvalidCloudContextstate
getLandingZone_cloudcontextMistmatch_throwsStaleconfig

Reply:
Since each test case is small, and a single step + assert, adding it into multiple individual tests would invariably add to maintenance overhead, hence retaining as a single function

Reply:
i.e. minimal value add in this particular case

Reply:
https://www.baeldung.com/java-unit-testing-best-practices#7-specific-unit-tests

https://www.baeldung.com/java-unit-testing-best-practices#3-test-case-naming-convention

Reply:
Agree with with general guidance, imo that makes sense when the unit tests have a little more meat in them i.e functionality being tested. 
It would be a single line in each test here, and don't see significant concerns in keeping it as a single function. 
Additionally since there are others tests in WSM in this pattern, this single function would not be out of place, hence at this time prefer to keep it this way. 

There is no explicit per-test setup. There is always a hidden overhead built in to any new test or function call - however minimal it is. Java (and high level languages) do a good job at optimization and hence developers don't typically bother about this.